packages/server/src/api/controllers/static/index.ts

```diff
@@ -287,6 +293,37 @@ export const getSignedUploadURL = async function (ctx: Ctx) {
     } catch (error: any) {
       ctx.throw(400, error)
     }
+  } else if (datasource?.source === "GOOGLE_CLOUD") {
+    const { bucket, key } = ctx.request.body || {}
+
+    const parsedKey = datasource?.config?.privateKey
+      .split(String.raw`\n`)
+      .join("\n")
```

Comment:
What's the purpose of these two lines? Maybe it's me being naive, but don't they cancel each other out?

Reply:
I ran into [this](https://stackoverflow.com/questions/74131595/error-error1e08010cdecoder-routinesunsupported-with-google-auth-library) issue when trying to authenticate. Turned out that the value from the google key gets escaped when stored in the db.

This code splits on escaped (`\\n`) new lines and replaces them before we use them. `String.raw` actually escapes the newline value passed in, turning `\n` into `\\n`. This is kind of opaque. The other example from that post is definitely clearer.

Example:

```javascript
// An string with an escaped new line 
let test = "Hi\\nthere!"

test.split(String.raw`\n`)
// String.raw escaped `\n` becoming `\\n`
// The string splits on the escaped value ['Hi', 'there!']

test.split(String.raw`\n`).join('\n')
// Finally becomes 'Hi\nthere!'
```

Reply:
Ewwww, that's not fun 😞